src/index.js

```diff
@@ -458,6 +460,12 @@ class CdiscountContentScript extends ContentScript {
         'yyyy-MM-dd'
       )}_Cdiscount_${amount}${currency}.pdf`
       const oneBill = {
+        shouldReplaceFile: (file, entry) => {
```

Comment:
For this, you have to be sure that the fileurl is stable. If, for whatever reason, cdiscount changes the url of its files, all the files will be downloaded one more time.

Maybe you should add the type of document in the entry (proof, bill) and juste compare the type